includes/class-wc-payments-order-success-page.php

```diff
@@ -28,39 +28,84 @@ public function __construct() {
 	 */
 	public function register_payment_method_override() {
 		// Override the payment method title on the order received page.
-		add_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woopay_payment_method_name' ], 10, 2 );
+		add_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woocommerce_payments_payment_method_name' ], 10, 2 );
 	}
 
 	/**
 	 * Remove the hook to override the payment method name on the order received page before the order summary.
 	 */
 	public function unregister_payment_method_override() {
-		remove_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woopay_payment_method_name' ], 10 );
+		remove_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woocommerce_payments_payment_method_name' ], 10 );
 	}
 
 	/**
-	 * Add the WooPay logo and the last 4 digits of the card used to the payment method name
-	 * on the order received page.
+	 * Hooked into `woocommerce_order_get_payment_method_title` to change the payment method title on the
+	 * order received page for WooPay and BNPL orders.
 	 *
-	 * @param string            $payment_method_title the default payment method title.
-	 * @param WC_Abstract_Order $abstract_order the order being shown.
+	 * @param string $payment_method_title Original payment method title.
+	 * @param WC_Abstract_Order $abstract_order Successful received order being shown.
+	 * @return string
 	 */
-	public function show_woopay_payment_method_name( $payment_method_title, $abstract_order ) {
-
+	public function show_woocommerce_payments_payment_method_name( $payment_method_title, $abstract_order ) {
 		// Only change the payment method title on the order received page.
 		if ( ! is_order_received_page() ) {
 			return $payment_method_title;
 		}
 
 		$order_id = $abstract_order->get_id();
 		$order    = wc_get_order( $order_id );
-		if ( ! $order || ! $order->get_meta( 'is_woopay' ) ) {
+
+		if ( ! $order ) {
+			return $payment_method_title;
+		}
+
+		$payment_method_id = $order->get_payment_method();
+
+		if ( stripos( $payment_method_id, 'woocommerce_payments' ) !== 0 ) {
+			return $payment_method_title;
+		}
+
+		if ( ! $order ) {
+			return $payment_method_title;
+		}
+
```

Comment:
```suggestion
```
I think we can remove this duplicate check.

Reply:
Resolved in 81b506b1c